test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -177,6 +182,66 @@ public void initClient() throws IOException {
         assert hasXPack != null;
         assert nodeVersions != null;
     }
+    
+    // Helper class to check warnings in REST responses with sensitivity to versions
+    // used in the target cluster.
+    protected class VersionSensitiveWarningsHandler implements WarningsHandler {
```

Comment:
I *think* it'd be nicer if this were `public` and `static`, maybe in its own file. There are places were we can't use `expectVersionSpecificWarnings` and there are places where we want to make calls that aren't subclasses of `ESTestCase`. We have precident for this, specifically making `client()` `public` and `static`.

Reply:
I think it'd be nice to have a small unit test for this.

Reply:
OK - I'll need to pass Set<Version> into the constructor as this is currently an inner class with direct access to EsRestTestCase.nodeVersions.

Reply:
Hmmmm. I'd be ok with a `public static SortedSet<Version> nodeVersions()` on `ESRestTestCase`. What do you think?

Reply:
I overlooked the fact nodeVersions is already static. See following comment below on EsRestTestCase and VersionSpecificWarningsHandler dependencies